app/src/main/java/org/schabi/newpipe/database/Migrations.java

```diff
@@ -197,6 +198,43 @@ public void migrate(@NonNull final SupportSQLiteDatabase database) {
         }
     };
 
+    public static final Migration MIGRATION_6_7 = new Migration(DB_VER_6, DB_VER_7) {
+        @Override
+        public void migrate(@NonNull final SupportSQLiteDatabase database) {
+            // Create a new column thumbnail_stream_id
+            database.execSQL("ALTER TABLE `playlists` ADD COLUMN `thumbnail_stream_id` "
+                    + "INTEGER NOT NULL DEFAULT -1");
+
+            // Migrate the thumbnail_url to the thumbnail_stream_id
+            database.execSQL("UPDATE playlists SET thumbnail_stream_id = ("
+                    + " SELECT CASE WHEN COUNT(*) != 0 then stream_uid ELSE -1 END"
+                    + " FROM ("
+                    + " SELECT p.uid AS playlist_uid, s.uid AS stream_uid"
+                    + " FROM playlists p"
+                    + " LEFT JOIN playlist_stream_join ps ON p.uid = ps.playlist_id"
+                    + " LEFT JOIN streams s ON s.uid = ps.stream_id"
+                    + " WHERE s.thumbnail_url = p.thumbnail_url) AS temporary_table"
+                    + " WHERE playlist_uid = playlists.uid)");
+
+            // Remove the thumbnail_url field in the playlist table
+            database.execSQL("CREATE TABLE IF NOT EXISTS `playlists_new`"
+                    + "(uid INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "
+                    + "name TEXT, "
+                    + "is_thumbnail_permanent INTEGER NOT NULL, "
+                    + "thumbnail_stream_id INTEGER NOT NULL)");
```

Comment:
Shouldn't `thumbnail_stream_id` be a foreign key to the streams table?

Reply:
The `thumbnail_stream_id` can be `-1` to indicate a default thumbnail and not always reference a stream. Therefore, IMO it isn't necessary to use it as a foreign key to the streams table.